app/controllers/users_controller.rb

```diff
@@ -172,6 +172,11 @@ def confirm_change_email
 
     @new_email = params[:new_email]
 
+    if @new_email.downcase == @user.email.downcase
+      flash[:error] = t("users.confirm_change_email.same_as_current")
+      render :change_email and return
+    end
```

Comment:
Very minor: this should probably also go under the comment about case sensitivity. 

Other than that, I think this PR is fine!